lib/fog/compute/google/requests/insert_instance_template.rb

```diff
@@ -0,0 +1,23 @@
+module Fog
+  module Compute
+    class Google
+      class Mock
+        def insert_instance_template(_name, _properties, _description)
+          Fog::Mock.not_implemented
+        end
+      end
+
+      class Real
+        def insert_instance_template(name, properties, description)
+          instance_template = ::Google::Apis::ComputeV1::InstanceTemplate.new(
+            :description => description,
+            :name => name,
+            :properties => properties,
```

Comment:
If I'm reading this correctly, `properties` will need to be an instance of google api client's `InstanceProperties` class.

If so - fog avoids exposing underlying API structures, so parameters need to be provided in hashes, see: https://github.com/fog/fog-google/blob/985cd021bcfe4f403deb7c0077997529d529b4b1/lib/fog/compute/google/requests/insert_server.rb